rust_dev_preview/dynamodb/src/bin/add-item.rs

```diff
@@ -3,12 +3,28 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-use aws_config::meta::region::RegionProviderChain;
-use aws_sdk_dynamodb::model::AttributeValue;
-use aws_sdk_dynamodb::{Client, Error, Region, PKG_VERSION};
-use std::process;
+use std::fmt::Display;
+
+use aws_sdk_dynamodb::{Client, Error};
+use dynamodb_code_examples::{
+    make_config,
+    scenario::create::{add_item, Item},
+    Opt as BaseOpt,
+};
 use structopt::StructOpt;
 
+#[derive(Debug)]
+struct PermissionError {
+    p_type: String,
+}
+
+impl std::error::Error for PermissionError {}
+impl Display for PermissionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "\n{} is not a valid permission type.\nYou must specify a permission type value of 'admin' or 'standard_user':\n-p PERMISSION-TYPE\n", self.p_type)
```

Comment:
It should be possible to line-wrap this for easier reading:
```suggestion
        write!(f, 
            "\n{} is not a valid permission type.\n\
            You must specify a permission type value of 'admin' or 'standard_user':\n\
            -p PERMISSION-TYPE\n",
            self.p_type
        )
```
The `\` will automatically strip any indentation.